feat: backfill tetris and pride noodles into the archive (#2863)
npmx-dev/npmx.dev: CONTRIBUTING.md

- `pnpm storybook` may log warnings or non-breaking errors for Nuxt modules due to the lack of mocks. If the UI renders correctly, these can be safely ignored.
- Do not `import type` from `.vue` files. The `vue-docgen-api` parser used by `@storybook/addon-docs` cannot follow type imports across SFCs and will crash. Extract shared types into a separate `.ts` file instead.

## Adding a noodle

"Noodles" are the occasional themed npmx logos that celebrate a day or event (Pride Month, World Tetris Day, the Node.js birthday, …). A noodle has two homes:

- the **homepage**, where it temporarily replaces the logo while it is active, and
- the **`/noodles` archive**, where every past noodle lives on permanently with its own detail page.

The two are wired up independently, so shipping a noodle to the homepage **and** the archive in the same PR keeps things in sync. Do both at once.

### 1. Add the logo component

> Create `app/components/Noodle/<Name>/Logo.vue`. Wrap the artwork in `TooltipApp` so hovering reveals the occasion, and reference poster art from `public/extra/`:

Reviewer: Not necessary. It might be a custom component, gif or whatever

```suggestion
Create `app/components/Noodle/<Name>/Logo.vue`. 
It's recommended to add a tooltip with event information to each noodle. If the poster is an image, add it to the `public/extra` directory. If the image differs between light and dark modes - prefer `ColorSchemeImg` component for poster
```

```vue
<template>
  <TooltipApp interactive position="top">
    <template #content>
      <p class="text-sm font-medium text-fg mb-1"><strong>World Tetris Day</strong></p>
    </template>
    <img
      width="400"
      class="w-80 sm:w-140 max-w-full mx-auto"
      src="/extra/tetris.svg"
      :alt="$t('alt_logo')"
    />
  </TooltipApp>
</template>
```

Drop the image(s) the logo points at into `public/extra/` (SVG preferred; PNG is fine).

> 

Reviewer: webp, gif, video - everything is fine if it match dynamic background

```suggestion

```

### 2. Register the logo and show it on the homepage

In `app/components/Noodle/index.ts`:

- import the component, and
- add it to the homepage rotation — `ACTIVE_NOODLES` for a date-bound noodle, or `PERMANENT_NOODLES` for one shown only behind a query param (e.g. `?kawaii`), and
- register it in the `NOODLE_LOGOS` map so the archive can resolve it by `key`.

```ts
import NoodleTetrisLogo from './Tetris/Logo.vue'

export const ACTIVE_NOODLES: Noodle[] = [
  {
    key: 'tetris',
    logo: NoodleTetrisLogo,
    date: '2026-06-06',
    dateTo: '2026-06-08',
    timezone: 'auto', // visitor's local time; or an IANA name like 'America/Los_Angeles'
    tagline: false, // hide the npmx tagline while active
  },
]

const NOODLE_LOGOS: Record<string, Component> = {
  // …
  tetris: NoodleTetrisLogo,
}
```

> > [!IMPORTANT]
> > The `key` here must exactly match the `key` of the archive entry you add in the next step — that is how the archive looks up the logo.

Reviewer: ```suggestion
> [!IMPORTANT]
> The `date` and `dateTo` keys are inclusive, meaning they specify the start (at 00:00) and end (at 23:59) dates. If the dates overlap, a noodle will be randomly selected on each visit.

> [!IMPORTANT]
> The `key` here must exactly match the `key` of the archive entry you add in the next step — that is how the archive looks up the logo.
```
